include/overlay005/bdhc.h

```diff
@@ -32,4 +35,15 @@ typedef struct BDHC {
     int stripsSize;
 } BDHC;
 
+BDHCUtil_CalculateObjectHeight(const fx32 objectHeight, const fx32 objectX, const fx32 objectZ, const BDHC *bdhc, fx32 *newObjectHeight);
+
+void BDHCLoader_Load(NARC *narc, const int bdhcSize, BDHC *bdhc, u8 *buffer);
+SysTask *BDHCLoader_StartTask(NARC *landDataNARC, const int param1, BDHC *bdhc, int *param3, u8 **buffer, int *param5);
+void BDHCLoader_ForceExitTask(SysTask *sysTask);
+void BDHCLoader_MarkBDHCNotLoaded(BDHC *bdhc);
```

Comment:
I think that we can simplify these names now:

```suggestion
void BDHC_Load(NARC *narc, const int bdhcSize, BDHC *bdhc, u8 *buffer);
SysTask *BDHC_LazyLoad(NARC *landDataNARC, const int param1, BDHC *bdhc, int *param3, u8 **buffer, int *param5);
void BDHC_KillLoad(SysTask *sysTask);
void BDHC_MarkNotLoaded(BDHC *bdhc);
```

Looking at the undocumented fields in `LazyLoad`, I also see that...

- `param1` (and `unk_E4`) is unused internally.
- `param3` (and `unk_DC`) is effectively unused (it might have been intended to be a communication mechanism to the client?).
- `param5` (and `unk_E8`) is effectively unused (it acts as a delayed kill-switch in the task's `Prepare` stage).

I think that we should update those names while we're in this module, just to capstone them.

Reply:
Agreed for `unk_E4`.

For `unk_E8`, it looks like it is used. Referenced [here](https://github.com/pret/pokeplatinum/blob/f1ac42f3875bd4345d3a985dd296d4bc458f1756/src/overlay005/ov5_021E779C.c#L503) (L503) as `param6->unk_10.unk_00`, increased right [above that line](https://github.com/pret/pokeplatinum/blob/f1ac42f3875bd4345d3a985dd296d4bc458f1756/src/overlay005/ov5_021E779C.c#L497) (L497) and passed right after (L498) to a function that starts a systask.  
This function + the systask itself are pretty much identical to our Load/LazyLoad; pretty certain that this is loading the NSBMD seeing the NNSG3d* types in the function parameters and seeing that it's started before the BDHC systask.
Anyway, `param6->unk_10.unk_00` is [now](https://github.com/pret/pokeplatinum/blob/f1ac42f3875bd4345d3a985dd296d4bc458f1756/src/overlay005/ov5_021E779C.c#L1939) (L1939) `v1->unk_20`, and in the actual systask routine, it's updated [here](https://github.com/pret/pokeplatinum/blob/f1ac42f3875bd4345d3a985dd296d4bc458f1756/src/overlay005/ov5_021E779C.c#L1917) (L1917) at the last subtask. (and just thought about it when I was writing this last sentence - should have probably called them "task steps")
So it looks like `unk_E8` is some sort of synchonization variable that forces the NSBMD(?) to be loaded before the BDHC.

Didn't bother checking if the same was true for `unk_DC`, but it may be another sync variable (they're [both in the same struct](https://github.com/pret/pokeplatinum/blob/f1ac42f3875bd4345d3a985dd296d4bc458f1756/src/overlay005/ov5_021E779C.c#L29-L32) in `ov5_021E779C`)

Reply:
And another question - what do you usually call unused variables? I guess `unusedX`?

Reply:
For parameters, just `unusedX` is fine. For struct fields, typically we use `dummyX` (where `X` is a byte-address within the struct, for clarity).

Reply:
Renamed the unused internal variables. Leaving `unk_DC`/`unk_E8` (and related) untouched for now as we'll need to explore what those pointers are shared with.